src/models/activity_pub/actor.cr

```diff
@@ -733,10 +788,28 @@ module ActivityPub
         "summary" => dig?(json, "https://www.w3.org/ns/activitystreams#summary", "und"),
         "icon" => dig_id?(json, "https://www.w3.org/ns/activitystreams#icon", "https://www.w3.org/ns/activitystreams#url"),
         "image" => dig_id?(json, "https://www.w3.org/ns/activitystreams#image", "https://www.w3.org/ns/activitystreams#url"),
-        "urls" => dig_ids?(json, "https://www.w3.org/ns/activitystreams#url")
+        "urls" => dig_ids?(json, "https://www.w3.org/ns/activitystreams#url"),
+        "attachments" => attachments_from_ldjson(
+          json.dig?("https://www.w3.org/ns/activitystreams#attachment")
+        )
       }.compact
     end
 
+    def self.attachments_from_ldjson(entry)
```

Comment:
once changes to @context are in place, the keys here will be in their expanded form. e.g.
```
    def self.attachments_from_ldjson(entry)
      entry_not_nil = (entry.try(&.as_a) || [] of JSON::Any).not_nil!

      entry_not_nil.reduce([] of Attachment) do |memo, a|
        name = dig?(a, "https://www.w3.org/ns/activitystreams#name", "und")
        type = a.dig?("@type").try(&.as_s)
        value = dig?(a, "http://schema.org#value")

        unless name.nil? || name.empty? || type.nil? || type.empty? || value.nil? || value.empty?
          value = maybe_wrap_link(value)
          memo << Attachment.new(name, type, value)
        end
        memo
      end
    end
```
"type" becomes "@type" and it's value will be "http://schema.org#PropertyValue"

Reply:
@toddsundsted I'm obviously doing something wrong. I added the schema info to the jsonld in the test 
```
  let(json) do
    <<-JSON
      {
        "@context":[
          "https://www.w3.org/ns/activitystreams",
          "https://w3id.org/security/v1",
          {
            "schema":"http://schema.org#",
            "PropertyValue": "schema:PropertyValue",
            "value": "schema:value"
          }
        ],
...
```


and used the code above. But I am not getting data that looks like that. Inside the loop in `self.attachments_from_ldjson` I'm still getting e.g.:

```
{"name" => "Blog", "type" => "PropertyValue", "value" => "https://somewhere.example.com"}
```

Reply:
ok, i'll take a look. there are probably a couple changes that have to be made at the same time. in the meantime, i pushed the changes that i made to [a branch](https://github.com/toddsundsted/ktistec/tree/add-attachments) on github so you could see what i did when i was working on this. this worked against mastodon (but not against your build because of the missing @context information).

relevant commit: https://github.com/toddsundsted/ktistec/commit/24a63f0fe28162f15ff1f6f0250207dcdc7c1d91

Reply:
(it's possible you have to undo the change to json_ld.cr as well)

Reply:
🤦 Missed that I had not removed that change to `json_ld.cr`. The ActivityPub stuff is all new to me.